data_model/src/asset.rs

```diff
@@ -302,91 +338,159 @@ pub struct Id {
-        AssetDefinition::new(id, AssetValueType::Quantity, true)
-    }
+    #[allow(missing_docs)]
+    impl WithId {
+        #[must_use]
+        #[inline]
+        pub fn quantity(self) -> WithMandatoryFields {
+            WithMandatoryFields {
+                id: self.id,
+                value_type: AssetValueType::Quantity,
+                mintable: Mintable::Infinitely,
+                metadata: Metadata::default(),
+            }
+        }
 
-    /// Token definition with quantity asset value type.
-    #[inline]
-    pub fn new_quantity_token(
-        id: <AssetDefinition as Identifiable>::Id,
-    ) -> <Self as Identifiable>::RegisteredWith {
-        AssetDefinition::new(id, AssetValueType::BigQuantity, true)
-    }
+        #[must_use]
+        #[inline]
+        pub fn big_quantity(self) -> WithMandatoryFields {
+            WithMandatoryFields {
+                value_type: AssetValueType::BigQuantity,
+                ..self.quantity()
+            }
+        }
 
-    /// Asset definition with big quantity asset value type.
-    #[inline]
-    pub fn new_big_quantity(
-        id: <AssetDefinition as Identifiable>::Id,
-    ) -> <Self as Identifiable>::RegisteredWith {
-        AssetDefinition::new(id, AssetValueType::BigQuantity, true)
+        #[must_use]
+        #[inline]
+        pub fn fixed(self) -> WithMandatoryFields {
+            WithMandatoryFields {
+                value_type: AssetValueType::Fixed,
+                ..self.quantity()
+            }
+        }
+
+        pub fn store(self) -> WithMandatoryFields {
+            WithMandatoryFields {
+                value_type: AssetValueType::Store,
+                ..self.quantity()
+            }
+        }
     }
 
-    /// Token definition with big quantity asset value type.
-    #[inline]
-    pub fn new_bin_quantity_token(
-        id: <AssetDefinition as Identifiable>::Id,
-    ) -> <Self as Identifiable>::RegisteredWith {
-        AssetDefinition::new(id, AssetValueType::BigQuantity, false)
+    impl WithMandatoryFields {
+        /// Make the builder create a [`Mintable::Once`] instance of [`AssetDefinition`]
+        #[must_use]
+        #[inline]
+        pub fn mintable_once(mut self) -> Self {
+            self.mintable = Mintable::Once;
+            self
+        }
+
+        /// Change mintability in-place
+        #[inline]
+        pub fn mintable(&mut self, mintable: bool) {
+            self.mintable = if mintable {
+                Mintable::Infinitely
+            } else {
+                Mintable::Once
+            };
+        }
+
+        /// Add metadata to builder
+        #[must_use]
+        #[inline]
+        pub fn with_metadata(mut self, metadata: Metadata) -> Self {
+            self.metadata = metadata;
+            self
+        }
+
+        /// Build [`AssetDefinition`]
+        #[must_use]
+        #[inline]
+        pub fn build(self) -> <AssetDefinition as Identifiable>::RegisteredWith {
```

Comment:
usually, builders have `mutable_api` flag because clients are submitting builders they never actually build the entity. We don't have to follow this convention here necessarily

Reply:
I view the presence of a builder as no more than a shorthand (if that). 

> clients are submitting builders they never actually build the entity. 

Technically `NewAssetDefinition` is the exception... Since the two are identical in terms of fields, I think there's no difference.